src/settingsdialog.cpp

```diff
@@ -0,0 +1,109 @@
+
+    ui->lineEditDebugPaths->setPlaceholderText( QLatin1String("auto-detect"));
+    ui->lineEditDebugPaths->setToolTip(QLatin1String("List of colon-separated paths that contain debug information."));
+
+    ui->lineEditKallsyms->setPlaceholderText(QLatin1String("auto-detect"));
+    ui->lineEditKallsyms->setToolTip(QLatin1String("Path to the kernel symbol mapping."));
+
+    ui->comboBoxArchitecture->setToolTip(QLatin1String("System architecture, e.g. x86_64, arm, aarch64 etc."));
+    ui->comboBoxArchitecture->setCurrentIndex(0);
+
+    connect(ui->btnSysroot, &QAbstractButton::clicked, this, [this] () {chooseDirectoryAndCopyToLineEdit(ui->lineEditSysroot);});
+    connect(ui->btnApplicationPath, &QAbstractButton::clicked, this, [this] () {chooseDirectoryAndCopyToLineEdit(ui->lineEditApplicationPath);});
+    connect(ui->btnExtraLibraryPaths, &QAbstractButton::clicked, this, [this] () {chooseDirectoryAndCopyToLineEdit(ui->lineEditExtraLibraryPaths);});
+    connect(ui->btnDebugPaths, &QAbstractButton::clicked, this, [this] () {chooseDirectoryAndCopyToLineEdit(ui->lineEditDebugPaths);});
+    connect(ui->btnKallsyms, &QAbstractButton::clicked, this, [this] () {chooseDirectoryAndCopyToLineEdit(ui->lineEditKallsyms);});
+
+    connect(ui->buttonBox, &QDialogButtonBox::clicked, this, &SettingsDialog::on_buttonBox_clicked);
+}
+
+SettingsDialog::~SettingsDialog() = default;
+
+void SettingsDialog::initSettings(const QString &sysroot, const QString &appPath, const QString &extraLibPaths,
+                                  const QString &debugPaths, const QString &kallsyms, const QString &arch,
+                                  const QString &objdump) {
+    ui->lineEditSysroot->setText(sysroot);
+    ui->lineEditApplicationPath->setText(appPath);
+    ui->lineEditExtraLibraryPaths->setText(extraLibPaths);
+    ui->lineEditDebugPaths->setText(debugPaths);
+    ui->lineEditKallsyms->setText(kallsyms);
+    ui->lineEditObjdump->setText(objdump);
+    int itemIndex = ui->comboBoxArchitecture->findText(arch);
+    // If index is -1 then arch from MainWindow class instance isn't in the combobox
+    if (0 <= itemIndex && itemIndex < ui->comboBoxArchitecture->count())
+        // Restore the text of current item in combobox by index
+        ui->comboBoxArchitecture->setCurrentIndex(itemIndex);
+}
+
+void SettingsDialog::chooseDirectoryAndCopyToLineEdit(QWebStyleEdit* qWebStyleEdit)
+{
+    QString sDir = QFileDialog::getExistingDirectory(nullptr, tr("Choose directory"), tr(""));
+    if (!sDir.isEmpty())
+        qWebStyleEdit->setText(sDir);
+}
+
+void SettingsDialog::on_buttonBox_clicked(QAbstractButton * button)
+{
+    // If "Ok" button pressed save each LineEdit text to MainWindow class instance variable
+    if (button == ui->buttonBox->button(QDialogButtonBox::Ok)) {
+        // Save LineEdit texts to the MainWindow class instance variables
+        emit sysrootChanged(ui->lineEditSysroot->displayText());
```

Comment:
add getters for the settings instead of changed signals, then use something like this in the mainwindow where the dialog is shown:

- forward the signals from the button box to the dialog: https://doc.qt.io/qt-5/qdialogbuttonbox.html#signals
- then listen to the `accepted` signal of the dialog in the mainwindow and set the settings there, i.e.:
```
connect(m_settingsDialog, &QDialog::accepted, this, [this]() {
    setAppPath(m_settingsDialog->appPath());
    ...
});
```

Reply:
Could you clarify should these getters have prefix 'get'?

Reply:
no